src/ristretto/ristretto_keys.rs

```diff
@@ -250,6 +267,36 @@ impl RistrettoPublicKey {
     }
 }
 
+//---------------------------------------   Ristretto Hashing Applications  ------------------------------------------//
+
+/// The Domain Separation Tag type for the KDF algorithm, version 1
+pub struct RistrettoKdf;
+impl DerivedKeyDomain for RistrettoKdf {
+    type DerivedKeyType = RistrettoSecretKey;
+}
+
+impl DomainSeparation for RistrettoKdf {
+    fn version() -> u8 {
+        1
+    }
+
+    fn domain() -> &'static str {
+        "com.tari.kdf.ristretto"
+    }
+}
+
+pub struct RistrettoGroupGenerator;
+
+impl DomainSeparation for RistrettoGroupGenerator {
```

Comment:
In some places we use Groups and some places PublicKey, should we not use one term for both?

Reply:
Yeah, historical naming from the PoC.  Even in the MW context, don't we refer to `G`, `H` etc as generator points? Maybe `RistrettoGeneratorPoint`  is the best name